### PR TITLE
upgrade to ruby 2.4.1

### PR DESCRIPTION
**Pull request by patmbolger**
PT Story: https://www.pivotaltracker.com/story/show/147050603

Ready for review:
@weedySeaDragon @robertcram @thesuss

### CODE REVIEW

File: .ruby-version [hunk 1]
```diff
@@ -1 +1 @@
-2.3.1
+ruby-2.4.1
```
Comment (weedySeaDragon) [inline .ruby-version]: I think you need to remove `ruby`; this should just be the version numbers, yes?

Comment (patmbolger) [reply .ruby-version]: Actually, this file is entirely created by RVM.

Comment (weedySeaDragon) [reply .ruby-version]: Yes, but this file is also used by `rbenv` and so it should really just contain the version numbers.  (The production system uses `rbenv`, BTW.)
You can have RVM put just the version numbers in there or you can create it manually. 
(Note that the content you're replacing only has the version numbers in it.)

Comment (patmbolger) [reply .ruby-version]: Yes, rbenv also supports the use of `.ruby-version`.  Does the presence of the "ruby-" string break rbenv? (if you use rbenv in your local env could you check that?)

**Review by RobertCram — APPROVED** [review]

File: .ruby-version (hunk 1 above)
Comment (RobertCram) [reply .ruby-version]: How did the .ruby-version file even get here? You put it in .gitignore in (already merged) PR https://github.com/AgileVentures/shf-project/pull/273

Comment (weedySeaDragon) [reply .ruby-version]: re: what is allowed in the `.ruby-version` file

Short answer:  `.rbenv` expects only the _version number_ in the file.  "ruby-" can't be in there.

### tldr:
rbenv expects the contents to match the 'name' of one of the ruby versions that it installs.   [This is the list of 'names' of possible versions that can be installed](https://github.com/rbenv/ruby-build/tree/master/share/ruby-build)
(tldr: the list is provided via `ruby-build`).
